controllers/windowsmachine/windowsmachine_controller.go

```diff
@@ -184,28 +182,16 @@ func (m *nodeToMachineMapper) Map(object handler.MapObject) []reconcile.Request
 		return nil
 	}
 
-	// Map the Node to the associated Machine through the Node's UID
-	machines := &mapi.MachineList{}
-	err := m.client.List(context.TODO(), machines,
-		client.MatchingLabels(map[string]string{MachineOSLabel: "Windows"}))
-	if err != nil {
-		log.Error(err, "could not get a list of machines")
-	}
-	for _, machine := range machines.Items {
-		if machine.Status.NodeRef != nil && machine.Status.NodeRef.UID == object.Meta.GetUID() {
-			return []reconcile.Request{
-				{
-					NamespacedName: types.NamespacedName{
-						Namespace: machine.GetNamespace(),
-						Name:      machine.GetName(),
-					},
-				},
-			}
-		}
+	// Get the Machine name through the Node's Machine annotation. The value has the form <namespace>/<name>
```

Comment:
I am hesitant to depend on a label. Have you looked at what other projects are doing or asked the SDK team for suggestions for this scenario?